### PR TITLE
#9970 url arr bug

### PR DESCRIPTION
**Pull request by Igi-ID**
## Description

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Bugfix

## Issue

**What is the current behavior?**
[#9970 ](https://github.com/geosolutions-it/MapStore2/issues/9970)
**What is the new behavior?**
Under scenarios described in the issue above, the export tool works and the layer is downloaded.
Under same scenarios, the style selector tab works as well.

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

File: web/client/utils/URLUtils.js [hunk 1]
```diff
@@ -111,3 +111,15 @@ export const isValidURLTemplate = (url, params, regexp = /^(http(s{0,1}):\/\/)+?
...
+
+/**
+ * Helper for working with a single string url.
+ * Use when calling implementations that do not know about array of urls, such as the `urlUtil` library,
+ * while still supporting our implementation of domain aliases and domain sharding.
+ *
+ * @param {string || array} url - Either a string representing a valid url or an array of strings which are all valid urls.
+ * @returns {string} Returns the argument if the argument is string, otherwise if the argument is an array, returns the first element.
+ */
+export const getDefaultUrl = (url) => {
```
Comment (MV88) [inline web/client/utils/URLUtils.js:123]: this requires a unit test, please add it

**Review by MV88 — CHANGES_REQUESTED** [review]
issue is resolved, please add the missing unit tests since the changes touches many parts from the core of mapstore

**Comment by Igi-ID** [comment]
> issue is resolved, please add the missing unit tests since the changes touches many parts from the core of mapstore

Tests included in https://github.com/geosolutions-it/MapStore2/pull/10004/commits/fff6b93adf9e6bf115e90aa79cdb1084210bda1d and https://github.com/geosolutions-it/MapStore2/pull/10004/commits/d85c2b914eaa07ec4a87151a91c3057e954f8988

File: web/client/api/WFS.js [hunk 2]
```diff
@@ -10,9 +10,10 @@ import urlUtil from 'url';
 import assign from 'object-assign';
 import requestBuilder from '../utils/ogc/WFS/RequestBuilder';
 import {toOGCFilterParts} from '../utils/FilterUtils';
+import { getDefaultUrl } from '../utils/URLUtils';
 
 export const toDescribeURL = (url, typeName) => {
-    const parsed = urlUtil.parse(url, true);
+    const parsed = urlUtil.parse(getDefaultUrl(url), true);
```
[the remark this answers is not in the document]
Comment (MV88) [reply web/client/api/WFS.js:16]: missing

File: web/client/api/CSW.js [hunk 3]
```diff
@@ -160,7 +161,7 @@ export const getLayerReferenceFromDc = (dc, options, checkEsri = true) => {
         const refs = castArray(dc.references);
         const wms = head(refs.filter((ref) => { return ref.scheme && REGEX_WMS_EXPLICIT.some(regex => ref.scheme.match(regex)); }));
         if (wms) {
-            let urlObj = urlUtil.parse(wms.value, true);
+            let urlObj = urlUtil.parse(getDefaultUrl(wms.value), true);
```
[the remark this answers is not in the document]
Comment (MV88) [reply web/client/api/CSW.js:174]: missing

File: web/client/observables/__tests__/wfs-test.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2020, GeoSolutions Sas.
```
Comment (MV88) [inline web/client/observables/__tests__/wfs-test.js:2]: new files should contain current year, please apply this to all new files
```suggestion
 * Copyright 2024, GeoSolutions Sas.
```

File: web/client/epics/__tests__/styleeditor-test.js [hunk 5]
```diff
@@ -1475,5 +1475,48 @@ describe('Test styleeditor epics, with mock axios', () => {
...
+        };
+        const NUMBER_OF_ACTIONS = 1;
+
+        const results = (actions) => {
+            expect(actions.length).toBe(NUMBER_OF_ACTIONS);
+            try {
+                actions.map((action) => {
+                    switch (action.type) {
+                    case LOADING_STYLE:
+                        expect(action.status).toBe('global');
```
Comment (MV88) [inline web/client/epics/__tests__/styleeditor-test.js:1502]: this is a false positive, i do not think this is actually testing new added code.

**Review by MV88 — CHANGES_REQUESTED** [review]
much better, but there are a few things missing.
please see inline comments

File: web/client/observables/__tests__/wfs-test.js (hunk 4 above)
Comment (Igi-ID) [reply web/client/observables/__tests__/wfs-test.js]: Included in https://github.com/geosolutions-it/MapStore2/pull/10004/commits/22f0d9ba262e3352d0572dcd6c748162626449d7

File: web/client/epics/__tests__/styleeditor-test.js (hunk 5 above)
Comment (Igi-ID) [reply web/client/epics/__tests__/styleeditor-test.js]: Adjusted test to better check the actual value in https://github.com/geosolutions-it/MapStore2/pull/10004/commits/22f0d9ba262e3352d0572dcd6c748162626449d7

File: web/client/api/WFS.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (Igi-ID) [reply web/client/api/WFS.js:16]: Added missing tests in https://github.com/geosolutions-it/MapStore2/pull/10004/commits/22f0d9ba262e3352d0572dcd6c748162626449d7

File: web/client/api/CSW.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (MV88) [reply web/client/api/CSW.js:174]: I do not think it is necessary

**Review by MV88 — APPROVED** [review]

**Comment by ElenaGallo** [comment]
test passed, @Igi-ID please backport to 2024.01.xx. Thanks

**Pull request merged**